dlpy/tensorboard.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python
+# encoding: utf-8
+#
+# Copyright SAS Institute
+#
+#  Licensed under the Apache License, Version 2.0 (the License);
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+''' TensorBoard class provides functionality for viewing scalar metrics in TensorBoard '''
+
+import os
+import swat
+
+class TensorBoard():
+    '''
+    Parameters
+    ----------
+    model : dlpy.Model
+        Specifies the desired model object to monitor.
+    log_dir : string
+        Specifies the directory to write logs to.
+    use_valid : bool
+        Specifies whether to record validation statistics.
+        If set to True then user must pass valid_table to Model.fit().
+        Default: False
+
+    Returns
+    -------
+    :class:`TensorBoard`
+
+    '''
+    def __init__(self, model, log_dir, use_valid=False):
+        self.model = model
+        if os.path.exists(log_dir):
+            self.log_dir = log_dir
+        else:
+            raise OSError(log_dir + " does not exist. Please provide an existing directory to write event logs or create this directory.")
+        self.use_valid = use_valid
+
+        # Scalar metrics to log
+        self.scalars = ['learning_rate', 'loss', 'error']
+        if self.use_valid:
+            self.scalars.append('valid_' + self.scalars[1])
+            self.scalars.append('valid_' + self.scalars[2])
+
+    def build_summary_writer(self):
+        '''
+        Creates a SummaryWriter object for logging scalar events 
```

Comment:
There should be a one-line description first.  Then a longer description after if needed.